options_test.go

```diff
@@ -2,2130 +2,149 @@ package dhcp6
+			desc: "value not present in Options map",
 			options: Options{
-				OptionClientArchType: [][]byte{{0, 5, 0, 9, 0, 0}},
-			},
-			arch: ArchTypes{
-				ArchTypeIntelLeanClient,
-				ArchTypeEFIx8664,
-				ArchTypeIntelx86PC,
+				2: [][]byte{[]byte("foo")},
 			},
-			ok: true,
-		},
-	}
-
-	for i, tt := range tests {
-		arch, ok, err := tt.options.ClientArchType()
-		if err != nil {
-			if want, got := tt.err, err; want != got {
-				t.Fatalf("[%02d] test %q, unexpected error for Options.ClientArchType: %v != %v",
-					i, tt.desc, want, got)
-			}
-
-			continue
-		}
-
-		if want, got := len(tt.arch), len(arch); want != got {
-			t.Fatalf("[%02d] test %q, unexpected arch slice length: %v != %v",
-				i, tt.desc, want, got)
-		}
-
-		for j := range arch {
-			if want, got := tt.arch[j], arch[j]; !reflect.DeepEqual(want, got) {
-				t.Fatalf("[%02d:%02d] test %q, unexpected value for Options.ClientArchType()\n- want: %v\n-  got: %v",
-					i, j, tt.desc, want, got)
-			}
-		}
-
-		if want, got := tt.ok, ok; want != got {
-			t.Fatalf("[%02d] test %q, unexpected ok for Options.ClientArchType(): %v != %v",
-				i, tt.desc, want, got)
-		}
-	}
-}
-
-// TestOptionsNII verifies that Options.NII properly parses and returns a
-// Network Interface Identifier value, if it is available with OptionNII.
-func TestOptionsNII(t *testing.T) {
-	var tests = []struct {
-		desc    string
-		options Options
-		nii     *NII
-		ok      bool
-		err     error
-	}{
-		{
-			desc: "OptionNII not present in Options map",
+			key: 1,
+			err: ErrOptionNotPresent,
 		},
 		{
-			desc: "OptionNII present in Options map, but too short length",
+			desc: "value present in Options map, but zero length value for key",
 			options: Options{
-				OptionNII: [][]byte{{1, 2}},
+				1: [][]byte{},
 			},
-			err: io.ErrUnexpectedEOF,
+			key: 1,
 		},
 		{
-			desc: "OptionNII present in Options map, but too long length",
+			desc: "value present in Options map",
 			options: Options{
-				OptionNII: [][]byte{{1, 2, 3, 4}},
+				1: [][]byte{[]byte("foo")},
 			},
-			err: io.ErrUnexpectedEOF,
+			key:   1,
+			value: []byte("foo"),
 		},
 		{
-			desc: "OptionNII present in Options map",
+			desc: "value present in Options map, with multiple values",
 			options: Options{
-				OptionNII: [][]byte{{1, 2, 3}},
-			},
-			nii: &NII{
-				Type:  1,
-				Major: 2,
-				Minor: 3,
+				1: [][]byte{[]byte("foo"), []byte("bar")},
 			},
-			ok: true,
+			key: 1,
+			err: ErrInvalidPacket,
 		},
 	}
 
 	for i, tt := range tests {
-		nii, ok, err := tt.options.NII()
-		if err != nil {
-			if want, got := tt.err, err; want != got {
-				t.Fatalf("[%02d] test %q, unexpected error for Options.NII(): %v != %v",
-					i, tt.desc, want, got)
-			}
-
+		value, err := tt.options.GetOne(tt.key)
+		if want, got := tt.err, err; want != got {
+			t.Errorf("[%02d] test %q, unexpected err for Options.GetOne(%v): %v != %v",
+				i, tt.desc, tt.key, want, got)
+		} else if err != nil {
```

Comment:
I think this is cleaner without the `else if` block, and more obvious that all error handling is occurring at the next level of indentation.